sgtm/artemis/get_document_from_firestore.tpl

```diff
@@ -0,0 +1,248 @@
+    "type": "TEXT",
+    "name": "gcpProjectId",
+    "displayName": "GCP Project ID (Where Firestore is located)",
+    "simpleValueType": true,
+    "notSetText": "projectId is retrieved from the environment variable GOOGLE_CLOUD_PROJECT",
+    "help": "Google Cloud Project ID where the Firestore database with margin data is located, leave empty to read from GOOGLE_CLOUD_PROJECT environment variable",
+    "canBeEmptyString": true
+  },
+  {
+    "type": "TEXT",
+    "name": "collectionId",
+    "displayName": "Firestore Collection ID",
+    "simpleValueType": true,
+    "defaultValue": "products",
+    "help": "The collection in Firestore that contains the products"
+  },
+  {
+    "type": "TEXT",
+    "name": "documentID",
+    "displayName": "Firestore Document ID",
+    "simpleValueType": true
+  },
+  {
+    "type": "CHECKBOX",
+    "name": "zeroIfNotFound",
+    "checkboxText": "Zero if not found",
+    "simpleValueType": true,
+    "defaultValue": true,
+    "help": "If true items that cannot be found in Firestore will be 0. If false items that cannot be found in Firestore will have their original value from the event data.",
+    "alwaysInSummary": true
+  }
+]
+
+
+___SANDBOXED_JS_FOR_SERVER___
+
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/**
+ * @fileoverview sGTM variable tag that retrieves a document from Firestore
+ * @version 1.0.0
+ */
+
+const Firestore = require("Firestore");
+const Promise = require("Promise");
+const getEventData = require("getEventData");
+const logToConsole = require("logToConsole");
+const makeString = require("makeString");
+const getType = require("getType");
+const JSON = require('JSON');
+
+/**
+ * Use Firestore to determine what the new value should be for this item.
+ * @param {String} item - documentID to retrieve from Firestore
+ */
+function getFirestoreDocument(documentId) {
+  
+  let document;
+
+  if (!documentId) {
+    logToConsole("No document Id");
+    return document;
+  }
+
+  const path = data.collectionId + "/" + documentId;
+
+  let firestore = Firestore;
+  if (getType(Firestore) === "function"){
+    firestore = Firestore();
+  }
+
+  return Promise.create((resolve) => {
+    return firestore.read(path, { projectId: data.gcpProjectId })
+    .then((result) => {
+      if (result.data != null) {
+        document = result.data;
+      } else {
+        logToConsole("Firestore document " + data.documentId + " does not exist");
```

Comment:
Does this fire? Does it not raise an error?
This else statement might be redundant.